### DISCUSSION TITLE
The heatmap & output from FAST can it be imported to DSA?

### DISCUSSION THREAD

**Discussion by lomshabhishek** [andreped/FP-dsa-plugin#15] [Ideas]
I was wondering if someone can import and visualise results from certain  project in FAST into DSA

**Comment by andreped** [comment]
> I was wondering if someone can import and visualise results from certain  project in FAST into DSA

Hello, @lomshabhishek! Hope you are doing well!

This is something I have been meaning to look into myself. In practice I do not believe it should be that difficult to do, but there are some caveats. First of all, the format used in FAST/FP when storing the output varies between models. For pw-classifiers we store the result in HDF5 format. For pw-emantic segmentation we store the result as pyramidal TIFFs.

DSA uses a different way of storing annotations, through GeoJSON, if I recall correctly, which is similar to what QuPath does.

Thus, to import annotations Into DSA from FP, requires that you convert the annotation first.

In the FP-DSA plugin, when you run a FAST model we actually do the full ordinary FAST pipeline and then convert the result. Thus to only import the result from FAST/FP into DSA, we could simply add a method that does the last part of the plugin. But we should try to support both HDF5 and TIFFs from the get-go.

Sadly this often tends to take more time actually developing than I anticipate, and I do not have a laptop where FAST in Docker works atm, but I may have later today. Let's keep this Discussion open and I will update you on any progress :]
